cmd/qat_plugin/README.md

```diff
@@ -52,81 +224,84 @@ The number of devices discovered are:10
+#### DPDK Prerequisites
 
-### Build QAT device plugin Docker image:
-```
-$ cd $GOPATH/src/github.com/intel/intel-device-plugins-for-kubernetes
-$ make  intel-qat-plugin
-```
+For the DPDK QAT demo to work, the Data Plane Development Kit (DPDK) drivers must be loaded and configured.
+For more information, refer to:
+[DPDK Getting Started Guide for Linux](https://doc.dpdk.org/guides/linux_gsg/index.html) and
+[DPDK Getting Started Guide, Linux Drivers section](http://dpdk.org/doc/guides/linux_gsg/linux_drivers.html)
 
-**Note**: `kerneldrv` mode is excluded from the build by default. Add `BUILDTAGS=kerneldrv` to `make` to get `kerneldrv` functionality added to the build.
+#### Build the image
 
-### Deploy QAT device plugin as a DaemonSet:
-```
-$ cd $GOPATH/src/github.com/intel/intel-device-plugins-for-kubernetes
-kubectl create -f deployments/qat_plugin/qat_plugin_default_configmap.yaml
-kubectl create -f deployments/qat_plugin/qat_plugin.yaml
+First you need to build the DPDK demo image:
+
+```bash
+$ cd $GOPATH/src/github.com/intel/intel-device-plugins-for-kubernetes/demo
+$ ./build-image.sh crypto-perf
+...
+Successfully tagged crypto-perf:devel
 ```
 
-**Note**: It is also possible to run the QAT device plugin using a non-root user. To do this,
-the nodes' DAC rules must be configured to allow PCI driver unbinding/binding, device plugin
-socket creation and kubelet registration. Furthermore, the deployments `securityContext` must
-be configured with appropriate `runAsUser/runAsGroup`.
+#### Deploy the pod
 
-### Verify QAT device plugin is registered on master:
-```
-$ kubectl describe node <node name> | grep qat.intel.com/generic
- qat.intel.com/generic: 10
- qat.intel.com/generic: 10
-```
+In the pod specification file, add container resource request and limit.
+For example, `qat.intel.com/generic: <number of devices>` for a container requesting QAT devices.
 
-### Consuming QAT device plugin
+For a DPDK-based workload, you may need to add hugepage request and limit.
 
-1. Build the DPDK image:
+```bash
+$ cd $GOPATH/src/github.com/intel/intel-device-plugins-for-kubernetes
+$ kubectl apply -k deployments/qat_dpdk_app/base/
+$ kubectl get pods
+  NAME                     READY     STATUS    RESTARTS   AGE
+  qat-dpdk                 1/1       Running   0          27m
+  intel-qat-plugin-5zgvb   1/1       Running   0          3h
 
-     ```
-     $ cd demo
-     $ ./build-image.sh crypto-perf
-     ```
+```
+
+> **Note**: The deployment example above uses kustomize that is available in
+> kubectl since Kubernetes v1.14 release.
 
-     This command produces a Docker image named `crypto-perf`.
+#### Run the test
```

Comment:
I've taken this demo to the next level but kept the original functionality so that this README is still valid and I did not have to make big changes to it.  ;)

It is also possible to something like

```
$ kubectl apply -k deployments/qat_dpdk_app/test-crypto1
$ kubectl apply -k deployments/qat_dpdk_app/test-compress1
```

so that no manual bash commands inside the container are not needed. it's then possible to observer the results via `kubectl logs`

Reply:
I suggest, if you want, you could update with a future PR after this one has landed..

Reply:
sure